From ec6cd9ee4a55355fe1ae46f5d2b93611a588f95c Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:42 -0400
Subject: [infiniband] mlx4: Use IS_ENABLED(CONFIG_IPV6)

Message-id: <b4412669f489fc9277041eb662da903c8cec3f2c.1410425016.git.avadai@redhat.com>
Patchwork-id: 93624
O-Subject: [Patch RHEL7.1 V1 046/197] IB/mlx4: Use IS_ENABLED(CONFIG_IPV6)
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Roland Dreier <roland@purestorage.com>

...instead of testing defined(CONFIG_IPV6) || defined(CONFIG_IPV6_MODULE)

Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 27cdef637c25705b433d5c4deeef4cf8dcb75d6a)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/mlx4/main.c b/drivers/infiniband/hw/mlx4/main.c
index 9682f38..a115abc 100644
--- a/drivers/infiniband/hw/mlx4/main.c
+++ b/drivers/infiniband/hw/mlx4/main.c
@@ -1598,7 +1598,7 @@ static int mlx4_ib_inet_event(struct notifier_block *this, unsigned long event,
  return NOTIFY_DONE;
 }
 
-#if defined(CONFIG_IPV6) || defined(CONFIG_IPV6_MODULE)
+#if IS_ENABLED(CONFIG_IPV6)
 static int mlx4_ib_inet6_event(struct notifier_block *this, unsigned long event,
     void *ptr)
 {
@@ -1618,7 +1618,7 @@ static void mlx4_ib_get_dev_addr(struct net_device *dev,
      struct mlx4_ib_dev *ibdev, u8 port)
 {
  struct in_device *in_dev;
-#if defined(CONFIG_IPV6) || defined(CONFIG_IPV6_MODULE)
+#if IS_ENABLED(CONFIG_IPV6)
  struct inet6_dev *in6_dev;
  union ib_gid  *pgid;
  struct inet6_ifaddr *ifp;
@@ -1641,7 +1641,7 @@ static void mlx4_ib_get_dev_addr(struct net_device *dev,
   endfor_ifa(in_dev);
   in_dev_put(in_dev);
  }
-#if defined(CONFIG_IPV6) || defined(CONFIG_IPV6_MODULE)
+#if IS_ENABLED(CONFIG_IPV6)
  /* IPv6 gids */
  in6_dev = in6_dev_get(dev);
  if (in6_dev) {
@@ -2120,7 +2120,7 @@ static void *mlx4_ib_add(struct mlx4_dev *dev)
     goto err_notif;
    }
   }
-#if defined(CONFIG_IPV6) || defined(CONFIG_IPV6_MODULE)
+#if IS_ENABLED(CONFIG_IPV6)
   if (!iboe->nb_inet6.notifier_call) {
    iboe->nb_inet6.notifier_call = mlx4_ib_inet6_event;
    err = register_inet6addr_notifier(&iboe->nb_inet6);
@@ -2171,7 +2171,7 @@ err_notif:
    pr_warn("failure unregistering notifier\n");
   ibdev->iboe.nb_inet.notifier_call = NULL;
  }
-#if defined(CONFIG_IPV6) || defined(CONFIG_IPV6_MODULE)
+#if IS_ENABLED(CONFIG_IPV6)
  if (ibdev->iboe.nb_inet6.notifier_call) {
   if (unregister_inet6addr_notifier(&ibdev->iboe.nb_inet6))
    pr_warn("failure unregistering notifier\n");
@@ -2296,7 +2296,7 @@ static void mlx4_ib_remove(struct mlx4_dev *dev, void *ibdev_ptr)
    pr_warn("failure unregistering notifier\n");
   ibdev->iboe.nb_inet.notifier_call = NULL;
  }
-#if defined(CONFIG_IPV6) || defined(CONFIG_IPV6_MODULE)
+#if IS_ENABLED(CONFIG_IPV6)
  if (ibdev->iboe.nb_inet6.notifier_call) {
   if (unregister_inet6addr_notifier(&ibdev->iboe.nb_inet6))
    pr_warn("failure unregistering notifier\n");
-- 
1.7.1